### PR TITLE
feat(cdk,core)!: allow passing icons at both sides simultaneously in textfield components

### LINKED ISSUE #978: 🚀 - TuiPrimitiveTextfieldComponent should allow to pass icons at both sides simultaneously
### Which @taiga-ui/* package(s) are relevant/releated to the feature request?

core, kit

### Description

I faced with the problem when I need to pass a left icon to `combo-box` or `select`. My hands are tied because I can pass only the left or right icon to the `TuiPrimitiveTextfieldComponent` simultaneously but `combo-box` and `select` already pass an arrow as `iconContent` to `TuiPrimitiveTextfieldComponent`.

Suggestion: 
Remove the restriction of passing only one icon to the `TuiPrimitiveTextfieldComponent`.

### PR DESCRIPTION
**Pull request by vladimirpotekhin**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Closes #978

## What is the new behavior?

# BREAKING CHANGES, DON'T MERGE IN 2.x

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

### CODE REVIEW

**Review by splincode — APPROVED** [review]

**Comment by splincode** [comment]
image

will there be another `input` in the future? aka `iconRight`

**Comment by vladimirpotekhin** [comment]
> will there be another `input` in the future? aka `iconRight`

I think current icon + new iconLeft is enough

File: projects/core/components/primitive-textfield/primitive-textfield.component.ts [hunk 1]
```diff
@@ -83,6 +84,12 @@ export class TuiPrimitiveTextfieldComponent
     iconContent: PolymorpheusContent<TuiContextWithImplicit<TuiSizeS | TuiSizeL>> | null =
         null;
 
+    @Input()
+    @tuiDefaultProp()
+    iconLeftContent: PolymorpheusContent<
```
Comment (waterplea) [inline projects/core/components/primitive-textfield/primitive-textfield.component.ts:89]: Let's ditch `null` and use empty string as lack of icon.

File: projects/core/components/primitive-textfield/primitive-textfield.template.html [hunk 2]
```diff
@@ -44,10 +44,10 @@
     </div>
     <div class="t-content">
         <div
-            *ngIf="iconAlignLeft"
+            *ngIf="iconAlignLeft || iconLeftContent"
             polymorpheus-outlet
             class="t-icon t-icon_left"
-            [content]="iconContent"
+            [content]="iconAlignLeft ? iconContent : iconLeftContent"
```
Comment (waterplea) [inline projects/core/components/primitive-textfield/primitive-textfield.template.html:50]: iconLeftContent || iconContent

**Review by waterplea — COMMENTED** [review]
What do you think if we move `icon` and `iconLeft` to `tuiTextfieldIcon` and `tuiTextfieldIconLeft`? Looks like we just pass it through to `PrimitiveTextfield` all the time and controllers are already merged upwards so components like `Select` who have an icon built in can just override it with `TUI_ARROW` or any other icon they need to provide to underlying `PrimitiveTextfield`.

File: projects/core/components/primitive-textfield/primitive-textfield-options.ts [hunk 3]
```diff
@@ -1,9 +1,7 @@
 import {InjectionToken, ValueProvider} from '@angular/core';
```
Comment (waterplea) [inline projects/core/components/primitive-textfield/primitive-textfield-options.ts:1]: This file is now a bit ridiculous 😄

**Review by waterplea — APPROVED** [review]

File: projects/core/components/primitive-textfield/primitive-textfield-options.ts (hunk 3 above)
Comment (vladimirpotekhin) [reply projects/core/components/primitive-textfield/primitive-textfield-options.ts:1]: yeah 😄 but hope it can be useful for something else in the future

**Pull request merged**